x-pack/heartbeat/monitors/browser/project.go

```diff
@@ -127,14 +127,14 @@ func (p *Project) jobs() []jobs.Job {
 	isScript := p.projectCfg.Source.Inline != nil
 	if isScript {
 		src := p.projectCfg.Source.Inline.Script
-		j = synthexec.InlineJourneyJob(context.TODO(), src, p.Params(), p.StdFields(), p.extraArgs()...)
+		j = synthexec.InlineJourneyJob(context.TODO(), src, p.Params(), p.StdFields(), p.projectCfg.Timeout, p.extraArgs()...)
```

Comment:
I think it'd be cleaner and more idiomatic to use [`context.WithTimeout`](https://pkg.go.dev/context#WithTimeout) than pass the extra parameter. We've passed this `context.TODO()` for ages in anticipation of having a real timeout, so now may be the time.

Reply:
Good sugestion, changing that now